### PR TITLE
Prepare DepthMap volume slicing

### PR DESCRIPTION
**Pull request by griwodz**
## Description

This pull request cleans up the code in DepthMap before a follow-up pull request that changes the
memory layout and sequence of computation.
The individual commits in this PR make very limited changes.

## Features list

- removes a lot of unused code all over DepthMap
- removes the redundant 2x transpose operation on all loaded images
- introduce some additional header files
- reduces the dynamic heap allocation of control structures on the CPU side of DepthMap
- change StaticVector to create directories for when writing intermediate files

### CODE REVIEW

File: src/aliceVision/mvsUtils/ImagesCache.cpp [hunk 1]
```diff
@@ -153,10 +124,10 @@ Color ImagesCache::getPixelValueInterpolated(const Point2d* pix, int camId)
...
     const float vi = pix->y - static_cast<float>(yp);
 
-    const Color lu = img[getPixelId(xp,     yp,     camId)];
-    const Color ru = img[getPixelId(xp + 1, yp,     camId)];
-    const Color rd = img[getPixelId(xp + 1, yp + 1, camId)];
-    const Color ld = img[getPixelId(xp,     yp + 1, camId)];
+    const Color lu = img->at( xp  , yp   ); // img[getPixelId(xp,     yp,     camId)];
+    const Color ru = img->at( xp+1, yp   ); // img[getPixelId(xp + 1, yp,     camId)];
+    const Color rd = img->at( xp+1, yp+1 ); // img[getPixelId(xp + 1, yp + 1, camId)];
+    const Color ld = img->at( xp  , yp+1 ); // img[getPixelId(xp,     yp + 1, camId)];
```
Comment (fabiencastan) [inline src/aliceVision/mvsUtils/ImagesCache.cpp]: would be good to remove the comments

File: src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp [hunk 2]
```diff
@@ -59,60 +61,57 @@ SemiGlobalMatchingRc::SemiGlobalMatchingRc(bool doComputeDepthsAndResetTCams, in
...
         }
     }
 }
 
 SemiGlobalMatchingRc::~SemiGlobalMatchingRc()
 {
-    delete tcams;
     delete depths;
-    delete depthsTcamsLimits;
+    // delete depthsTcamsLimits;
```
Comment (fabiencastan) [inline src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp]: comments to remove

File: src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp [hunk 3]
```diff
@@ -29,25 +29,27 @@ namespace depthMap {
...
+    , step( _step )
 {
     sp = _sp;
 
-    rc = _rc;
-    scale = _scale;
-    step = _step;
+    // rc = _rc;
+    // scale = _scale;
+    // step = _step;
```
Comment (fabiencastan) [inline src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp]: comments to remove

**Comment by griwodz** [comment]
Comments are removed as proposed.

**Pull request merged**